Settings Auto-Renew: Use core components on the survey modal

gabrielcaires opened the pull request:
Depends ON #103964
Closes DOTCON-78

## Proposed Changes
* Use core components to build the survey modal

## Why are these changes being made?
* Style consistency 


Before
![image](https://github.com/user-attachments/assets/44ca9eb7-bc83-4323-b34c-e09243b4e166)


After
![CleanShot 2025-06-05 at 13 03 23@2x](https://github.com/user-attachments/assets/4db27758-738f-4c02-a1d2-5b768178924f)


## Testing Instructions

* Create a Business site
* Go to /me/purchases
* Add a valid credit card as a payment method
* Click on  `Auto-renew is ON` 
* Click `Turn off auto renewer` 
* Check if the modal was rendered as the screenshot above


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

gabrielcaires committed:
Add single confirmation dialog

gabrielcaires committed:
Use core confirm dialog

gabrielcaires committed:
Move component to shareable folder

gabrielcaires committed:
Add readme

gabrielcaires committed:
Footer spacing adjustment

gabrielcaires committed:
Use core elements

gabrielcaires committed:
Adjust chat button

andres-blanco reviewed (approved):
LGTM
<img width="641" alt="image" src="https://github.com/user-attachments/assets/be226184-a951-485f-8b14-40779b29302b" />

gabrielcaires committed:
Adjust copy

gabrielcaires committed:
Adjust modal font-size

gabrielcaires commented:
![CleanShot 2025-06-09 at 10 45 53@2x](https://github.com/user-attachments/assets/3e8dc137-753e-4a1c-8643-37a56d805e08)

> I like how we're confirming that auto-renew is off now but that doesn't work as a title. We also shouldn't include punctuation in titles. Instead, let's use "Help us improve" as a title so that it frames the content better.

✅ Fixed

> Let's incorporate that auto-renew has been turned off in the copy. I added "You've turned off auto-renewal." to the paragraph.

✅ Fixed


> We've been cleaning up the "Need help?" link in other modals to not include "Contact us".

✅ Fixed


> It kind of bugs me that the radio button copy is so small compared to the paragraph copy. 

I double-checked it and saw it using the Gutenberg font-size as described on the [component storybook](https://wordpress.github.io/gutenberg/?path=/docs/components-radiocontrol--docs). It is the same size we are describing on our figma file (WordPress-Design-System--Gutenberg)

I noticed the model content is using a possibly wrong" font-size (16px). I adjusted to 14px. You can see the adjusted version below. Should we go with this size?

![CleanShot 2025-06-09 at 11 04 10@2x](https://github.com/user-attachments/assets/fa281f33-3c29-4bf9-b9cd-13002f938159)


**Another topic:**

I still think is a bit weird mix `You've turned off auto-renewal`  + request to answer the question + question on the same text block.  I had difficulty reading all the copy.

Do you think we could split it?

gabrielcaires committed:
Merge branch 'trunk' into update/use-core-dialog-survey

gabrielcaires committed:
Set confirmation message to another paragraph

gabrielcaires commented:
![CleanShot 2025-06-10 at 11 47 06@2x](https://github.com/user-attachments/assets/f458f40e-4b8e-4f1e-a91d-798a3b29e645)
✅

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17552493

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @gabrielcaires for including a screenshot in the description! This is really helpful for our translators.
